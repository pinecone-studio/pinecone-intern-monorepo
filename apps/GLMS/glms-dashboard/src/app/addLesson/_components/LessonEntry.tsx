import React from 'react';
import { Input } from '@/components/ui/input';
import { Textarea } from '@/components/ui/textarea';

export interface InputData {
  topic: string;
  title: string;
  details: string;
}

interface LessonEntryProps {
  inputData: InputData;
  handleInputChange: (_field: keyof InputData) => (_e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
}

export const LessonEntry: React.FC<LessonEntryProps> = ({ inputData, handleInputChange }) => {
  return (
    <div className="flex flex-col pt-[64px] gap-[32px]">
      <div data-testid="continue topic">
        <p className="font-inter text-base font-semibold leading-5 text-[#121316] pb-[8px]">Topic</p>
        <Input data-testid="lesson_input_comp" className="w-[588px] border-gray-400" placeholder="Select" value={inputData.topic} onChange={handleInputChange('topic')} />
      </div>
      <div data-testid="continue title">
        <p className="font-inter text-base font-semibold leading-5 text-[#121316] pb-[8px]">Title</p>
        <Input data-testid="lesson_input_sec_comp" className="w-[588px] border-gray-400" placeholder="Enter here..." value={inputData.title} onChange={handleInputChange('title')} />
      </div>
      <div data-testid="continue details">
        <p className="font-inter text-base font-semibold leading-5 text-[#121316] pb-[8px]">Details</p>
        <Textarea
          data-testid="lesson_text_area_comp"
          className="w-[588px] h-[110px] max-h-[200px] border-gray-400"
          placeholder="Write here..."
          value={inputData.details}
          onChange={handleInputChange('details')}
        />
      </div>
    </div>
  );
};
