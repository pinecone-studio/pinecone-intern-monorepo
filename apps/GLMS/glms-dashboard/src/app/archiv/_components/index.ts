export * from './Card';
export * from './CardData';