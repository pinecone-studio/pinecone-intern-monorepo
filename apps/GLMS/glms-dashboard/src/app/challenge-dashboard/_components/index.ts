export * from './CoreButton';
export * from './RadioButton';
export * from './Select';
