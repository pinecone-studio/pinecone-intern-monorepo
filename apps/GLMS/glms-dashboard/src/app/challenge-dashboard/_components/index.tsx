export * from './RadioButton';
export * from './Select';
export * from './CoreButton';
export * from './SelectWithLabel';
export * from './Input';