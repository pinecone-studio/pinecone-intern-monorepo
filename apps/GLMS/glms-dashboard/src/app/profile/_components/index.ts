export * from './ProfileButton';
export * from './SignIn';
