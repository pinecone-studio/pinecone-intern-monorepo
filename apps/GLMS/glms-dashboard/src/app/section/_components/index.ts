export * from "./GetSections"
export * from "./SectionForm"