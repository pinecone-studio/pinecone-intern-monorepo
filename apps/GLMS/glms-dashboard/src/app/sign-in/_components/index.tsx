export * from './TextInput';
export * from './Loader';
