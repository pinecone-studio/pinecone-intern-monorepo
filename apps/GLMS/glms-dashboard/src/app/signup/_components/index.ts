export * from './Loader';
export * from './TextInput';