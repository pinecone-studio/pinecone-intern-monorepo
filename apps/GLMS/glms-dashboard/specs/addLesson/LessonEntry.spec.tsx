// LessonEntry.test.tsx
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { LessonEntry } from '../../src/app/addLesson/_components/LessonEntry';

// Mock Input and Textarea components
jest.mock('@/components/ui/input', () => ({
  Input: ({ className, placeholder, value, onChange }: never) => <input className={className} placeholder={placeholder} value={value} onChange={onChange} data-testid="input" />,
}));

jest.mock('@/components/ui/textarea', () => ({
  Textarea: ({ className, placeholder, value, onChange }: never) => <textarea className={className} placeholder={placeholder} value={value} onChange={onChange} data-testid="textarea" />,
}));

describe('LessonEntry Component', () => {
  const mockInputData = {
    title: 'Initial Title',
    topic: 'Initial Topic',
    details: 'Initial Details',
  };

  const mockSetInputData = jest.fn();

  const renderComponent = (inputData = mockInputData) => {
    return render(<LessonEntry inputData={inputData} setInputData={mockSetInputData} />);
  };

  afterEach(() => {
    jest.clearAllMocks(); // Reset mocks after each test
  });

  it('renders without crashing', () => {
    renderComponent();
    expect(screen.getByPlaceholderText('Сонгоно уу')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Оруулна уу...')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Энд бичнэ үү...')).toBeInTheDocument();
  });

  it('displays correct initial values', () => {
    renderComponent();
    expect(screen.getByPlaceholderText('Сонгоно уу')).toHaveValue(mockInputData.topic);
    expect(screen.getByPlaceholderText('Оруулна уу...')).toHaveValue(mockInputData.title);
    expect(screen.getByPlaceholderText('Энд бичнэ үү...')).toHaveValue(mockInputData.details);
  });

  it('calls setInputData with updated topic', () => {
    renderComponent();
    const newValue = 'Updated Topic';
    const input = screen.getByPlaceholderText('Сонгоно уу');
    fireEvent.change(input, { target: { value: newValue } });
    expect(mockSetInputData).toHaveBeenCalledTimes(1);
    // Simulate state update call
    const updater = mockSetInputData.mock.calls[0][0];
    expect(updater(mockInputData)).toEqual({
      ...mockInputData,
      topic: newValue,
    });
  });

  it('calls setInputData with updated title', () => {
    renderComponent();
    const newValue = 'Updated Title';
    const input = screen.getByPlaceholderText('Оруулна уу...');
    fireEvent.change(input, { target: { value: newValue } });
    expect(mockSetInputData).toHaveBeenCalledTimes(1);
    // Simulate state update call
    const updater = mockSetInputData.mock.calls[0][0];
    expect(updater(mockInputData)).toEqual({
      ...mockInputData,
      title: newValue,
    });
  });

  it('calls setInputData with updated details', () => {
    renderComponent();
    const newValue = 'Updated Details';
    const textarea = screen.getByPlaceholderText('Энд бичнэ үү...');
    fireEvent.change(textarea, { target: { value: newValue } });
    expect(mockSetInputData).toHaveBeenCalledTimes(1);
    // Simulate state update call
    const updater = mockSetInputData.mock.calls[0][0];
    expect(updater(mockInputData)).toEqual({
      ...mockInputData,
      details: newValue,
    });
  });

  it('matches snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
