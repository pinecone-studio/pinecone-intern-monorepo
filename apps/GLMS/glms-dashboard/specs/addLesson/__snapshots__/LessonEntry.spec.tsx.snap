// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonEntry Component matches snapshot 1`] = `
<div>
  <div>
    <div
      class="flex flex-col pt-[64px] gap-[32px]"
    >
      <div>
        <p
          class="font-inter text-base font-semibold leading-5 text-[#121316] pb-[8px]"
        >
          Сэдэв
        </p>
        <input
          class="w-[588px] border-gray-400"
          data-testid="input"
          placeholder="Сонгоно уу"
          value="Initial Topic"
        />
      </div>
      <div>
        <p
          class="font-inter text-base font-semibold leading-5 text-[#121316] pb-[8px]"
        >
          Гарчиг
        </p>
        <input
          class="w-[588px] border-gray-400"
          data-testid="input"
          placeholder="Оруулна уу..."
          value="Initial Title"
        />
      </div>
      <div>
        <p
          class="font-inter text-base font-semibold leading-5 text-[#121316] pb-[8px]"
        >
          Дэлгэрэнгүй
        </p>
        <textarea
          class="pb-[70px] w-[588px] h-[110px] max-h-[200px] border-gray-400"
          data-testid="textarea"
          placeholder="Энд бичнэ үү..."
        >
          Initial Details
        </textarea>
      </div>
    </div>
  </div>
</div>
`;
